### PR TITLE
FIT-286 Implement Semantic Versioning Support for Geo Services API

### PR DESCRIPTION
**Pull request by mariamcws**
## Description
Implement Semantic Versioning Support for Geo Services API

I still need to add deploy to pre-int and integration and update manifest changes once this is merged and working

## Jira Issue link

[Implement Semantic Versioning Support for Geo Services API](https://osi-cwds.atlassian.net/browse/FIT-286)

## Types of changes
- [x] New feature (non-breaking change for continuous integration)

### CODE REVIEW

**Comment by mariamcws** [comment]
retest this please

File: Jenkinsfile [hunk 1]
```diff
@@ -69,23 +80,26 @@ node ('tpt2-slave'){
...
            sleep (20)
-       }
-      stage('Integration Tests') {
-          git branch: '$branch', url: 'https://github.com/ca-cwds/geo-services-api.git'
-          buildInfo = rtGradle.run buildFile: 'build.gradle', tasks: 'integrationTest --stacktrace'
-          publishHTML([allowMissing: true, alwaysLinkToLastBuild: true, keepAll: true, reportDir: 'build/reports/tests/integrationTest', reportFiles: 'index.html', reportName: 'Integration Tests Reports', reportTitles: 'Integration tests summary'])
-         cleanWs()
-      }
+        }
+        stage('Integration Tests') {
```
Comment (LeonidMarushevskyi) [inline Jenkinsfile]: Is it temporary?

File: Jenkinsfile [hunk 2]
```diff
@@ -69,23 +80,26 @@ node ('tpt2-slave'){
...
-       publishHTML([allowMissing: true, alwaysLinkToLastBuild: true, keepAll: true, reportDir: 'build/reports/tests/test', reportFiles: 'index.html', reportName: 'JUnitReports', reportTitles: 'JUnit tests summary'])
-       cleanWs()
-      }
+//       emailext attachLog: true, body: "Failed: ${e}", recipientProviders: [[$class: 'DevelopersRecipientProvider']],
+//       subject: "GEO Services API CI pipeline failed", to: "Leonid.Marushevskiy@osi.ca.gov, Alex.Kuznetsov@osi.ca.gov, Oleg.Korniichuk@osi.ca.gov, alexander.serbin@engagepoint.com, vladimir.petrusha@engagepoint.com"
+//       slackSend channel: "#geo-services-api", baseUrl: 'https://hooks.slack.com/services/', tokenCredentialId: 'slackmessagetpt2', message: "GEO Services API pipeline failed: ${env.JOB_NAME} ${env.BUILD_NUMBER}"
+//       publishHTML([allowMissing: true, alwaysLinkToLastBuild: true, keepAll: true, reportDir: 'build/reports/tests/integrationTest', reportFiles: 'index.html', reportName: 'Integration Tests Reports', reportTitles: 'Integration tests summary'])
```
Comment (LeonidMarushevskyi) [inline Jenkinsfile]: Why did this commented?

Comment (edgibbs) [reply Jenkinsfile]: I'm assuming for testing purposes, but before merging you'd want to put it back.

File: Jenkinsfile (hunk 1 above)
Comment (mariamcws) [reply Jenkinsfile]: yes

File: Jenkinsfile (hunk 2 above)
Comment (mariamcws) [reply Jenkinsfile]: temporary for debug

File: Jenkinsfile [hunk 3]
```diff
@@ -30,37 +33,45 @@ node ('tpt2-slave'){
        rtGradle.useWrapper = true
    }
    stage('Build'){
-       def buildInfo = rtGradle.run buildFile: 'build.gradle', tasks: 'readArguments jar -DRelease=$RELEASE_PROJECT -DBuildNumber=$BUILD_NUMBER -DCustomVersion=$OVERRIDE_VERSION'
+       def buildInfo = rtGradle.run buildFile: 'build.gradle', tasks: "readArguments jar -DRelease=\$RELEASE_PROJECT -DBuildNumber=\$BUILD_NUMBER -DCustomVersion=\$OVERRIDE_VERSION".toString()
```
Comment (edgibbs) [inline Jenkinsfile]: So this is mostly just a question, but did any of these resolve correctly before when they were single quoted?  I experimented in the groovy console and it doesn't do interpolations for single quotes.

Comment (mariamcws) [reply Jenkinsfile]: the single quoted code did not have toString() so it resolved fine, to apply toString() we need double quotes

File: Jenkinsfile (hunk 1 above)
Comment (mariamcws) [reply Jenkinsfile]: it is back in now

File: Jenkinsfile (hunk 2 above)
Comment (mariamcws) [reply Jenkinsfile]: it is added back now

File: Jenkinsfile (hunk 3 above)
Comment (AVandana-OSI) [reply Jenkinsfile]: In this case we can use Single Quote

Comment (mariamcws) [reply Jenkinsfile]: actually, on testing, single quotes also resolve fine with toString()

**Pull request merged**